pumpkin/src/client/player_packet.rs

```diff
@@ -541,6 +541,13 @@ impl Player {
         }
     }
 
+    pub async fn handle_player_abilities(&self, player_abilities: SPlayerAbilities) {
+        let mut abilities = self.abilities.lock().await;
+
+        // Set the flying ability
+        abilities.flying = player_abilities.flags & 0x02 != 0 && abilities.allow_flying;
```

Comment:
The abilities have `allow_flying` as well, Should should set them and not check if they true in the packet

Reply:
I've seen that this packet is sent by the client only when the player starts/stops flying and is not sending the "allow_fly" property. I've implemented the same check that vanilla does (bit mask 0x02 on the Flags field of the packed and the additional check on the player "allow_fly" flag) to set the flying property

Reply:
Oh, Just looked at vanilla code and your right!, nvm then